src/components/Apps.tsx

```diff
@@ -69,26 +64,37 @@ export default class extends React.Component<any, any> {
   componentWillReceiveProps(nextProps: any) {
     const id = nextProps.match.params.id;
 
-    if (id && id !== this.props.match.params.id) {
-      this.fetchApp(id);
+    if (id !== this.props.match.params.id) {
+      if (id) {
+        this.fetchApp(id);
+      } else {
+        this.setState({ currentApp: null, currentGroups: null, currentUsers: null });
+      }
     }
   }
 
   render() {
     const currentApp = this.state.currentApp as any;
+    const appId = _.get(currentApp, 'id');
     const { currentUsers, currentGroups } = this.state;
 
     return (
       <div className={`row ${css(styles.container)}`}>
         <ListPane
           Component={App}
-          getKey={item => item.id}
           getData={getApps}
+          selectedItem={currentApp}
           onSelect={app => {
-            this.props.history.push(`/apps/${app.id}`);
+            if (app.id === appId) {
+              this.props.history.push(`/apps`);
+            } else {
+              this.props.history.push(`/apps/${app.id}`);
+            }
```

Comment:
Not for this PR, but I think we might need a refactor where the route change would drive the state (or some of it), instead of only reflecting state